lib/substituteScope.js

```diff
@@ -0,0 +1,24 @@
+function substituteScope(scope) {
```

Comment:
just copying from before so it's easier to keep track of what's left (no worries for not getting to it yet)

> can you name this file substituteScope.js to match its name? :) 
> also can you add a comment explaining the purpose of this function? I'm a bit confused about what it does - is it replacing scope things? Shouldn't that happen in the steps instead?